### PR TITLE
Fix node crash on multiple trasnactions

### LINKED ISSUE #8610: The "otherBuffer" argument must be an instance of Buffer or Uint8Array. Received undefined
### Expected behavior
At any condition, it shouldn't crash the node. It should not crash the node

### Actual behavior
```
2023-06-15T23:18:11.255Z INFO sample engine 3545012 [id=2130b72e7655c90c5c3dc8b78730a2f4198daef3f1d4f13b1c878f9b39b66049 height=251410 generator=lsk3oe7hmf5k5e4f58zhmbouzx3t4v98sgsfuhwnw numberOfTransactions=1 numberOfAssets=1 numberOfEvents=9] Block executed
2023-06-15T23:18:11.256Z INFO sample engine 3545012 [status=success event=chain_newBlock] IPCServer published event
2023-06-15T23:18:11.257Z INFO sample engine 3545012 [status=success event=network_newBlock] IPCServer published event
2023-06-15T23:18:11.266Z INFO sample engine 3545012 [status=success event=chain_newBlock] WSServer published event
2023-06-15T23:18:11.270Z INFO sample engine 3545012 [status=success event=network_newBlock] WSServer published event
2023-06-15T23:18:11.489Z FATAL sample application 3545012 [err=The "otherBuffer" argument must be an instance of Buffer or Uint8Array. Received undefined] System error: unhandledRejection
```

### Steps to reproduce
Unknown

### Which version(s) does this affect? (Environment, OS, etc...)
beta.1-

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #8610 

### How was it solved?

- Fix how transaction pool and generator communicates

### How was it tested?

```
const validators = require('./config/default/dev-validators.json');

const { createWSClient } = require('@liskhq/lisk-api-client');


(async () => {
	const client = await createWSClient('ws://127.0.0.1:7887/rpc-ws');

	const acct = await client.invoke('auth_getAuthAccount', { address: validators.keys[0].address });
	for (let i = 0; i < 10; i++) {
		const tx = await client.transaction.create({
			module: 'token',
			command: 'transfer',
			nonce: (BigInt(acct.nonce) + BigInt(i)).toString(),
			params: {
				amount: '50000000000000',
				recipientAddress: validators.keys[0].address,
				data: '',
				tokenID: '0400000000000000',
			},
			fee: 500000000,
		}, validators.keys[0].privateKey);
		console.log(tx);
		await client.transaction.send(tx);
	}

	await client.disconnect();
})();

```

### CODE REVIEW

File: elements/lisk-transaction-pool/src/transaction_pool.ts [hunk 1]
```diff
@@ -413,35 +384,57 @@ export class TransactionPool {
...
+						break;
+					}
+				} catch (error) {
+					veirfyResults.push({
+						id: transaction.id,
+						status: TransactionStatus.INVALID,
+						transaction,
+					});
+					break;
+				}
```
Comment (bobanm) [inline elements/lisk-transaction-pool/src/transaction_pool.ts:406]: It seems to me that this loop and the very next one `for (const tx of allTransactions)` could be combined in one?

**Review by bobanm — CHANGES_REQUESTED** [review]

File: elements/lisk-transaction-pool/src/transaction_pool.ts [hunk 2]
```diff
@@ -413,35 +384,57 @@ export class TransactionPool {
...
-			let firstInvalidTransaction: { id: Buffer; status: TransactionStatus } | undefined;
-			try {
-				await this._applyFunction(allTransactions);
-			} catch (error) {
-				const failedStatus = this._getStatus(error as TransactionFailedResponse);
-				firstInvalidTransaction = {
-					id: (error as TransactionFailedResponse).id,
-					status: failedStatus,
-				};
+			const veirfyResults = [];
```
Comment (ishantiw) [inline elements/lisk-transaction-pool/src/transaction_pool.ts:387]: also in other paces below
```suggestion
			const verifyResults = [];
```

File: elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts [hunk 3]
```diff
@@ -358,14 +353,14 @@ describe('TransactionPool class', () => {
...
 			const MAX_TRANSACTIONS = 10;
 			transactionPool = new TransactionPool({
-				applyTransactions: jest.fn(),
+				verifyTransaction: jest.fn(),
 				minEntranceFeePriority: BigInt(10),
 				maxTransactions: MAX_TRANSACTIONS,
 				maxPayloadLength: 15360,
 			});
 
 			const tempApplyTransactionsStub = jest.fn();
```
Comment (ishantiw) [inline elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts:362]: also in other places below
```suggestion
			const tempVerifyTransactionsStub = jest.fn();
```

File: elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts [hunk 4]
```diff
@@ -524,14 +515,14 @@ describe('TransactionPool class', () => {
...
```
Comment (ishantiw) [inline elements/lisk-transaction-pool/test/unit/transaction_pool.spec.ts:524]:
```suggestion
			const tempVerifyTransactionsStub = jest.fn();
```

File: elements/lisk-transaction-pool/src/types.ts [hunk 5]
```diff
@@ -33,8 +33,9 @@ export enum Status {
...
 
+// This enum is matching with framework/src/abi/constants.ts
 export enum TransactionStatus {
-	INVALID = 0,
-	UNPROCESSABLE,
-	PROCESSABLE,
+	INVALID = -1,
+	UNPROCESSABLE = 0,
+	PROCESSABLE = 1,
 }
```
Comment (ishantiw) [inline elements/lisk-transaction-pool/src/types.ts:41]: I think we can simply do,
```
export enum TransactionStatus {
	INVALID = -1,
	UNPROCESSABLE,
	PROCESSABLE,
}
```

**Review by bobanm — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]

**Pull request merged**